lib/JSON/Validator/OpenAPI/Dancer2.pm

```diff
@@ -0,0 +1,72 @@
+package JSON::Validator::OpenAPI::Dancer2;
+
+use Carp qw(confess);
+use Hash::MultiValue;
+use Mojo::Base 'JSON::Validator::OpenAPI';
+
+### request ###
+
+sub _get_request_data {
+  my ($self, $dsl, $in) = @_;
+
+  return $dsl->query_parameters->as_hashref_mixed if $in eq 'query';
+  return $dsl->route_parameters->as_hashref_mixed if $in eq 'path';
+  return $dsl->body_parameters->as_hashref_mixed  if $in eq 'formData';
+  return Hash::MultiValue->new($dsl->app->request->headers->flatten)->as_hashref_mixed
+    if $in eq 'header';
+  return $dsl->app->request->data if $in eq 'body';
+  confess "Unsupported \$in: $in. Please report at https://github.com/jhthorsen/json-validator";
+}
+
+sub _get_request_uploads {
+  my ($self, $dsl, $name) = @_;
+
+  return ($dsl->app->request->upload($name));    # context-aware
```

Comment:
Oh, the comment reminds that `upload()` is context-aware. I added the parens to enforce list context. See other comments about `($value) =  ...`